machine_test.go

```diff
@@ -1645,3 +1652,22 @@ func TestCreateSnapshot(t *testing.T) {
 		})
 	}
 }
+
+func testAttachBalloonDevice(ctx context.Context, t *testing.T, m *Machine){
+	if err := m.AttachBalloon(ctx, m.Cfg.BalloonDevice); err != nil {
+		t.Errorf("attaching balloon device failed from testAttachBalloon: %s", err)
+	}
+}
+
+func testGetBalloonConfig(ctx context.Context, t *testing.T, m *Machine){
+	config := models.Balloon {}
+
+	err := m.GetBalloonConfig(ctx, &config)
```

Comment:
Unless you have a good reason, it is better to return a value, instead of modifying the parameter (in this case, &config).

Reply:
The reason why I modify config, because how [GetBalloonConfig](https://github.com/RoyceDavison/firecracker-go-sdk/blob/fc55cf297893e99dcdf4aef475e5e12ee79a935f/machine.go#L1122) is implemented. I want to test whether it returns a correct value, so I use unmarshal to decoding the JSON and return a go struct. 

But if I understand correctly, do you mean I could just return a JSON value and maybe don't need unmarshal at all? JSON is ok in here(I think payloadData is JSON string)?

Reply:
But GetBalloonConfig is your function. Does it have to be like that?

Reply:
I followed the [GetMetadata](https://github.com/RoyceDavison/firecracker-go-sdk/blob/06a30fe6e3152e91471c60c3182b90dc66e1a0e1/machine.go#L915) to implement `GetBalloonConfig`. But I can return a balloon to avoid modifying the param.